app/workers/machine_translation_worker.rb

```diff
@@ -0,0 +1,41 @@
+class MachineTranslationWorker
+  include Sidekiq::Worker
+
+  def perform(target, author)
+    target = YAML::load(target)
+    author = YAML::load(author)
+    field = DynamicAnnotation::Field.joins(:annotation).where('annotations.annotation_type' => 'language', 'annotations.annotated_type' => target.class.name, 'annotations.annotated_id' => target.id.to_s, field_type: 'language').first
+    src_lang = field.value unless field.nil?
+    mt = target.annotations.where(annotation_type: 'mt').last
+    unless mt.nil? or src_lang.nil?
+      translations = []
+      text = target.text
+      # Assume languages format is ['en', 'ar', ...]
+      languages = target.project.settings[:languages] unless target.project.settings.nil?
```

Comment:
Same as above: I suggest to replace this line by languages = self.project.get_languages, which already handles if settings are not present (among other things) and makes the code simpler.